src/modules/Shadow/Module.js

```diff
@@ -0,0 +1,146 @@
+/*
+* Create Shadow
+*/
+module.exports = function canvasResize(options, UI) {
+
+  var defaults = require('./../../util/getDefaults.js')(require('./info.json'));
+  const pixelSetter = require('../../util/pixelSetter.js');
+
+  var output;
+
+  function draw(input, callback, progressObj) {
+
+    options.X_value = parseInt(options.X_value || defaults.X_value);
+    options.Y_value = parseInt(options.Y_value || defaults.Y_value);
+
+    progressObj.stop(true);
+    progressObj.overrideFlag = true;
+
+    var step = this;
+
+    function extraManipulation(pixels) {
+      let [w, h] = pixels.shape;
+      let newPixels = require('ndarray')(new Uint8Array(4 * (w + Math.abs(options.X_value)) * (h + Math.abs(options.Y_value))).fill(0), [(w + Math.abs(options.X_value)), (h + Math.abs(options.Y_value)), 4]);
+      let iMax = w,
+        jMax = h;
+      if (options.X_value < 0 && options.Y_value < 0) {
+        for (var k = 0; k < Math.abs(options.X_value); k++) {
+          for (var l = 0; l < (h + Math.abs(options.Y_value)); l++) {
+            let val = 255 - ((k / Math.abs(options.X_value)) * 255);
+            pixelSetter(k, l, [val, val, val, 255], newPixels);
+          }
+        }
+        for (var k = 0; k < (w + Math.abs(options.X_value)); k++) {
+          for (var l = 0; l < Math.abs(options.Y_value); l++) {
+            if (k < Math.abs(options.X_value) && k < l) {
+              continue;
+            }
+            let val = 255 - ((l / Math.abs(options.Y_value)) * 255);
+            pixelSetter(k, l, [val, val, val, 255], newPixels);
+          }
+        }
+        for (let i = 0; i < iMax && i < w; i++) {
+          for (let j = 0; j < jMax && j < h; j++) {
+            let x = i + Math.abs(options.X_value), y = j + Math.abs(options.Y_value);
+            pixelSetter(x, y, [pixels.get(i, j, 0), pixels.get(i, j, 1), pixels.get(i, j, 2), pixels.get(i, j, 3)], newPixels);
+          }
+        }
+      }
+
+      else if (options.X_value >= 0 && options.Y_value >= 0) {
```

Comment:
Looks like there is a lot of repetition here, can that be fixed?

Reply:
Do we feel this is blocking, or should we just open a new issue to address it? Thank you all! We could include this in `v3.7.0`!

Reply:
Hmm, this review is quite old. We didn't get any reply so ig we can dismiss it :)